rg_instructor_analytics/tests/test_view.py

```diff
@@ -0,0 +1,149 @@
+"""Test for view."""
+from datetime import date
+
+from django.http.request import QueryDict
+from django.test import RequestFactory, TestCase
+from mock import Mock, patch
+from opaque_keys import InvalidKeyError
+
+from rg_instructor_analytics.views import EnrollmentStatisticView
+
+
+class TestEnrollmentStatisticView(TestCase):
+    """Test for enrollment statistic view."""
+
+    MOCK_PREVIOUS_ENROLL_STATE = (4, -1)
+    MOCK_CURRENT_ENROLL_STATE = [
+        {'date': date(2017, 10, 6), 'count': 1, 'is_active': False},
+        {'date': date(2017, 10, 7), 'count': 1, 'is_active': False},
+        {'date': date(2017, 10, 8), 'count': 2, 'is_active': False},
+        {'date': date(2017, 10, 8), 'count': 1, 'is_active': True},
+        {'date': date(2017, 10, 14), 'count': 1, 'is_active': True},
+        {'date': date(2017, 10, 20), 'count': 1, 'is_active': True},
+        {'date': date(2017, 10, 23), 'count': 1, 'is_active': True},
+    ]
+    MOCK_FROM_DATE = 1507150800
+    MOCK_TO_DATE = 1509573600
+    MOCK_REQUEST_PARAMS = QueryDict('from=' + str(MOCK_FROM_DATE) + '&to=' + str(MOCK_TO_DATE))
+    MOCK_COURSE_ID = 'course-v1:test+course+id'
+    MOCK_ALLOWED_USER = 'staff'
+    MOCK_UN_ALLOWED_USER = 'honor'
+
+    EXPECTED_ENROLL_STATISTIC = {
+        'dates': [1507150800, 1507262400, 1507348800, 1507435200, 1507953600, 1508472000, 1508731200, 1509573600],
+        'total': [3, 2, 1, 0, 1, 2, 3, 3],
+        'enroll': [4, 4, 4, 5, 6, 7, 8, 8],
+        'unenroll': [-1, -2, -3, -5, -5, -5, -5, -5],
+    }
+
+    def setUp(self):
+        """Implement from base class."""
+        self.factory = RequestFactory()
+
+    @patch('rg_instructor_analytics.views.has_access')
+    @patch('rg_instructor_analytics.views.get_course_by_id')
+    @patch('rg_instructor_analytics.views.CourseKey.from_string')
+    @patch('rg_instructor_analytics.views.EnrollmentStatisticView.get_statistic_per_day')
+    def test_enroll_statistic_post_call(
+            self, moc_get_statistic_per_day,
+            moc_course_key_from_string,
+            moc_get_course_by_id,
+            moc_has_access):
+        """Verify standard post flow."""
+        moc_get_statistic_per_day.return_value = self.EXPECTED_ENROLL_STATISTIC
+        moc_course_key_from_string.return_value = 'key'
+        moc_get_course_by_id.return_value = 'course'
+        moc_has_access.return_value = True
+
+        request = self.factory.post('/courses/' + self.MOCK_COURSE_ID + '/tab/instructor_analytics/api/enroll_statics/')
```

Comment:
using format function is more efficient with multiple adding strings

Reply:
Done